rio_tiler_pds/sentinel/aws/sentinel2.py

```diff
@@ -248,7 +262,11 @@ def __attrs_post_init__(self):
         self.crs = WGS84_CRS
 
         self.bands = tuple(
-            [band for band in self.stac_item["assets"] if re.match("B[0-9A]{2}", band)]
+            [
+                sentinel_l2a_band_map.get(band)
+                for band in self.stac_item["assets"]
+                if sentinel_l2a_band_map.get(band)
+            ]
```

Comment:
We want to apply the mapping for the new 1.0 stac items but leave it alone for the existing 1.0.0-beta.2 stac items. But won't this remove any bands that are named `B**`? Because `sentinel_l2a_band_map.get("B01")` will return `None` and be skipped.

I think we can do this mapping more simply:
```py
self.bands = (
    sentinel_l2a_band_map.get(band, band)
    for band in self.stac_item["assets"]
)
```
so that we return the existing band name if it isn't found in the mapping.

The other worry I have here is that we should probably allow users to use the _new_ asset ids as well, because at some point in the future the archive will be fully transferred over I assume. If we _only_ apply the mapping, then we support the old names but not the new names.

This would be a slightly bigger change because we'd have to store the new STAC asset ids in `self.bands` and then apply the mapping when creating the s3 filename

Reply:
Good points. My assumption was that the whole archive will get updated, so once that is finished we can implement the mapping. I ran some random testing and for 98% of the results the bands were in the new format. The archive is of course massive, so potentially there are still a significant amount of scenes in the old format. 

As for your proposed solution, this is indeed simpler but it will also return bands which we do not want to return as available bands. Example:

```python
self.bands = (
    sentinel_l2a_band_map.get(band, band)
    for band in self.stac_item["assets"]
)

> ['aot', 'B02', 'B01', 'granule_metadata', 'B03', 'B08', 'B8A', 'B09', 'B04', 'B05', 'B06', 'B07', 'scl', 'B11', 'B12', 'thumbnail', 'tileinfo_metadata', 'visual', 'wvp', 'aot-jp2', 'blue-jp2', 'coastal-jp2', 'green-jp2', 'nir-jp2', 'nir08-jp2', 'nir09-jp2', 'red-jp2', 'rededge1-jp2', 'rededge2-jp2', 'rededge3-jp2', 'scl-jp2', 'swir16-jp2', 'swir22-jp2', 'visual-jp2', 'wvp-jp2']
```

We do not want to return bands like 'thumbnail' and 'aot' for example as these are not COG files.

Supporting the new band naming would probably be preferred in the future, but I do not know how big the impact will be on the applications using `rio-tiler-pds`. We can then do something like

```python
self.bands = (
    band for band in self.stac_item["assets"] if band in sentinel_l2a_band_map.keys()
)
> ['blue', 'coastal', 'green', 'nir', 'nir08', 'nir09', 'red', 'rededge1', 'rededge2', 'rededge3', 'swir16', 'swir22']
```

to only return the bands with actual COG files. Or of course just replace the mapping `dict` with a `list` of available COG bands.

Reply:
sorry for picking up this only now. Did we decide anything here? 

I'm planning to do a release soon